src/process/dust/ccpr_scheme_fengsha_mod.F90

```diff
@@ -34,36 +34,102 @@ module CCPr_Scheme_Fengsha_Mod
+      RDRAG,            &
+      TSKIN,            &
+      USTAR,            &
+      USTAR_THRESHOLD,  &
+      GWETTOP,          &
+      z0,               &
+      CLAYFRAC,         &
+      SANDFRAC,         &
+      AIRDEN,           &
+      FROCEAN,          &
+      FRLANDICE,        &
+      FRSNOW,           &
+      alpha,            &
+      gamma,            &
+      reff,             &
+      rlower,           &
+      rupper,           &
+      MoistOpt,         &
+      DragOpt,          &
+      HorizFluxOpt,     &
+      TotalEmission,    &
+      EmissionBin,      &
+      RC)
 
       ! Uses
       USE Constants,     Only : g0
       Use CCPr_Dust_Common_Mod
       use precision_mod, only : fp, ZERO
-      Use MetState_Mod,  Only : MetStateType
-      Use DiagState_Mod, Only : DiagStateType
-      Use Error_Mod,     Only : CC_SUCCESS, CC_FAILURE, CC_Error
-      Use CCPr_Dust_Common_Mod, Only : DustStateType
 
       IMPLICIT NONE
 
       ! Arguments
-      type(MetStateType),  intent(in)    :: MetState     !< Meteorological Variables
-      type(DiagStateType), intent(inout) :: DiagState    !< Diagnostic Variables
-      type(DustStateType), intent(inout) :: DustState    !< Dust Variables
-
-      integer, intent(out) :: RC                      !< Success or Failure
+      integer,  intent(in) :: nDustSpecies    !< Number of Dust Species
+      integer,  intent(in) :: DSOILTYPE       !< Dominant Soil Type
+      real(fp), intent(in) :: SSM             !< Sediment Supply Map (0-1)
+      real(fp), intent(in) :: RDRAG           !< Drag Partition (0-1)
+      real(fp), intent(in) :: TSKIN           !< Skin Temperature [K]
+      real(fp), intent(in) :: USTAR           !< Friction Velocity [m/s]
+      real(fp), intent(in) :: USTAR_THRESHOLD !< Friction Velocity Threshold [m/s]
+      real(fp), intent(in) :: CLAYFRAC        !< Fraction of clay (0-1)
+      real(fp), intent(in) :: SANDFRAC        !< Fraction of sand (0-1)
+      real(fp), intent(in) :: AIRDEN          !< Air Density [kg/m3]
+      real(fp), intent(in) :: alpha           !< Alpha Parameter
+      real(fp), intent(in) :: gamma           !< Gamma Parameter
+      real(fp), intent(in) :: FROCEAN          !< Fraction of Land
+      real(fp), intent(in) :: FRLANDICE       !< Fraction of Land Ice
+      real(fp), intent(in) :: FRSNOW          !< Fraction of Snow
+      real(fp), intent(in) :: GWETTOP         !< Top soil wetness [m3/m3]
+      real(fp), intent(in) :: z0              !< Surface roughness [m]
+
+      real(fp), dimension(:), intent(in) :: reff            !< Effective Radius
+      real(fp), dimension(:), intent(in) :: rlower          !< Lower Radius
+      real(fp), dimension(:), intent(in) :: rupper         !< Upper Radius
+
+      integer,  intent(in) :: MoistOpt        !< Option for moisture
+      integer,  intent(in) :: DragOpt         !< Option for drag
+      integer,  intent(in) :: HorizFluxOpt    !< Option for horizontal flux
```

Comment:
Still think you could make these `optional`

Reply:
made these optional